File: X/js/XRPL_UKL1/soal7_0.js
/*
Soal        : 7
Modelsoal   : 0
*/
const readline = require("readline");

const input = readline.createInterface({
    input: process.stdin,
    output: process.stdout
});

function hitung(nilai){
    bayar=2000;
    if (nilai>10){
        bayar=bayar+1000;
    }
    if (nilai>20){
        bayar=bayar+1000;
    }
    if (nilai>30){
        bayar=bayar+1000;
    }

    return nilai*bayar+10000;
}

namaPelanggan=["Ali", "Budi", "Dani", "Edi", "Umar"];//nama pelanggan 

input.question("Masukkan jumlah pelanggan : ", function(id) {
id=parseInt(id)-1;
//console.log("Masukkan jumlah pelanggan : 1");
//id = 1-1;

input.question("Masukkan jumlah pelanggan : ", function(tagihan) {
tagihan=parseInt(tagihan);
//console.log("Masukkan jumlah tagihan : 33");
//tagihan = 33;

tagihan=hitung(tagihan);

console.log(`
Print Out Tagihan
ID\t: ${id+1}
Nama\t: ${namaPelanggan[id]}
Tagihan\t: Rp.${tagihan},-
`)
input.close()
})})

File: X/rust/XRPL_UKL1/soal7_0.rs
use std::io::{self, Write};

const NAMA_PELANGGAN: [&str; 5] = ["Ali", "Budi", "Dani", "Edi", "Umar"];
const MAX_PELANGGAN: usize = NAMA_PELANGGAN.len();

fn hitung(nilai: i32) -> i32 {
    let mut bayar = 2000;
    if nilai > 10 {
        bayar += 1000;
    }
    if nilai > 20 {
        bayar += 1000;
    }
    if nilai > 30 {
        bayar += 1000;
    }
    nilai * bayar + 10000
}

fn input_int(msg: &str) -> i32 {
    print!("{}", msg);
    io::stdout().flush().unwrap();

    let mut input = String::new();
    io::stdin().read_line(&mut input).unwrap();
    input.trim().parse().unwrap()
}

fn main() {
    let id = input_int("Masukkan jumlah pelanggan: ") - 1;
    if (id < 0) || (id >= MAX_PELANGGAN as i32) {
        println!("ID pelanggan tidak valid!");
        return;
    }

    let tagihan = input_int("Masukkan jumlah tagihan: ");
    if tagihan < 0 {
        println!("Jumlah tagihan tidak valid!");
        return;
    }

    let tagihan = hitung(tagihan);

    println!(
        "\nPrint Out Tagihan\n\
        ID\t: {}\n\
        Nama\t: {}\n\
        Tagihan\t: Rp.{},-\n",
        id + 1,
        NAMA_PELANGGAN[id as usize],
        tagihan
    );
}
